### PR TITLE
[v1.10.0] [zos_copy] zos_copy migration to ZOAU v1.3.0

### LINKED ISSUE #1106: [Enabler] [ zos_copy] Migrate module, module utilities and test cases to ZOAU 1.3
### Is there an existing issue for this?

- [X] There are no existing issues.

### Ansible module

_No response_

### Is there an existing issue for this?

- [X] There are no existing issues.

### Ansible module

zos_archive

### Epic
[[Enabler] [Epic] Adopt new ZOAU 1.3 Python changes](https://github.com/ansible-collections/ibm_zos_core/issues/931)

### Enabler description

With the adoption of ZOAU 1.3, modules, module utilities and test cases will need to be updated according to the [migration guide](https://www.ibm.com/docs/en/zoau/1.3.x?topic=planning-migrating-zoau-v13-from-v12x-earlier). 

APIs changes impacting module:
    - Impacted by ZOAU Dataset API: Y
    - Impacted by ZOAU jobs API: N
    - Impacted rename of  types.py to ztypes.py: N
    - Impacted contents of module volumes.py moved to datasets.py: N
    - Impacted by opercmd timeout to milli sec: N
    - Impacted by JSON schema change: N
    - Impacted by Excecptions schema: N

Additional highlights:
TBD

### PR DESCRIPTION
**Pull request by rexemin**
##### SUMMARY
Fixes #1106.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
- zos_copy
- module_utils/dataset.py
- module_utils/job.py
- Tests for zos_job_submit that depend on zos_copy

### CODE REVIEW

**Comment by rexemin** [comment]
Results for the test suites of zos_copy and zos_job_submit (which had a couple of tests that depend on zos_copy).

Screenshot 2024-02-13 at 15 00 32
Screenshot 2024-02-13 at 15 01 28
Screenshot 2024-02-13 at 15 01 43

**Comment by rexemin** [comment]
Screenshot 2024-02-13 at 15 22 18
Pipeline results for the 2 tests that depended on zos_encode that I tested after its migration was merged.

Screenshot 2024-02-13 at 15 23 50
Screenshot 2024-02-13 at 15 24 00

File: plugins/modules/zos_copy.py [hunk 1]
```diff
@@ -918,7 +920,7 @@ def copy_to_seq(
             if self.force_lock:
                 copy_args["options"] += " -f"
 
-            response = datasets._copy(new_src, dest, None, **copy_args)
+            response = datasets._copy(new_src, dest, **copy_args)
```
Comment (fernandofloresg) [inline plugins/modules/zos_copy.py:923]: hey could you review if is possible to remove underscore zoau functions ? Not saying is wrong or bad but do remember Demetri said to remove if possible.

**Review by fernandofloresg — CHANGES_REQUESTED** [review]
Just a request to replace underscore functions. Your code and test changes are ok, If is not possible to remove underscore functions let me know and I will approve

File: plugins/modules/zos_copy.py (hunk 1 above)
Comment (rexemin) [reply plugins/modules/zos_copy.py]: Removed all calls to hidden ZOAU functions and ran the pipeline again for zos_copy.

Screenshot 2024-02-16 at 16 19 14
Screenshot 2024-02-16 at 16 19 55

Comment (rexemin) [reply plugins/modules/zos_copy.py]: And for zos_job_submit

Screenshot 2024-02-16 at 16 33 38
Screenshot 2024-02-16 at 16 33 53

**Review by fernandofloresg — APPROVED** [review]
This looks good to me.

**Comment by fernandofloresg** [comment]
This will also validate #1146

**Review by ketankelkar — APPROVED** [review]
LGTM

**Pull request merged**